### PR TITLE
[#187300092] [Set up Local tests & CI]

### PR DESCRIPTION
**Pull request by Nadinefiona**
## What does this PR do?

This PR sets up local tests and integrates all necessary Continuous Integration (CI) tools, including CodeClimate and CircleCI. It also adds badges to the README file.

## Description of Task to be completed?

The primary task accomplished in this PR is to establish a robust testing framework locally and integrate it with our CI pipeline. 

## How should this be manually tested?

this will be tested by running "npm run test "

## What are the relevant pivotal tracker/Trello stories?

- Trello ID: [#187300092]

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,24 +3,31 @@
...
-    "typescript": "^5",
+    "@types/chai": "^4.3.16",
+    "@types/mocha": "^10.0.6",
     "@types/node": "^20",
     "@types/react": "^18",
     "@types/react-dom": "^18",
+    "chai": "^5.1.1",
+    "eslint": "^8",
+    "eslint-config-next": "14.2.3",
+    "mocha": "^10.4.0",
```
Comment (fab-ryan) [inline package.json]: we use Jest for testing

Comment (fab-ryan) [inline package.json]: we use Jest for testing

**Review by fab-ryan — CHANGES_REQUESTED** [review]
You will use Jest testing not chai

**Review by fab-ryan — CHANGES_REQUESTED** [review]
add Single one test

**Review by fab-ryan — CHANGES_REQUESTED** [review]
add one test

**Comment by Nadinefiona** [comment]
> add one test



> add one test
sure,I am working on it.

**Review by fab-ryan — APPROVED** [review]

**Review by musabehonore — APPROVED** [review]

**Review by Sophie20202 — APPROVED** [review]

**Review by RWEMAREMY — APPROVED** [review]
good job

**Pull request merged**